controller/cache/cache.go

```diff
@@ -389,38 +392,66 @@ func (c *liveStateCache) Run(ctx context.Context) error {
 
 	go c.watchSettings(ctx)
 
-	kube.RetryUntilSucceed(func() error {
-		clusterEventCallback := func(event *db.ClusterEvent) {
-			c.lock.Lock()
-			cluster, ok := c.clusters[event.Cluster.Server]
-			if ok {
-				defer c.lock.Unlock()
-				if event.Type == watch.Deleted {
-					cluster.Invalidate()
-					delete(c.clusters, event.Cluster.Server)
-				} else if event.Type == watch.Modified {
-					cluster.Invalidate(clustercache.SetConfig(event.Cluster.RESTConfig()))
-				}
-			} else {
-				c.lock.Unlock()
-				if event.Type == watch.Added && isClusterHasApps(c.appInformer.GetStore().List(), event.Cluster) {
-					go func() {
-						// warm up cache for cluster with apps
-						_, _ = c.getSyncedCluster(event.Cluster.Server)
-					}()
-				}
-			}
-		}
-
-		return c.db.WatchClusters(ctx, clusterEventCallback)
-
-	}, "watch clusters", ctx, clustercache.ClusterRetryTimeout)
+	err = c.db.WatchClusters(ctx, c.namespace, c.handleAddEvent, c.handleModEvent, c.handleDeleteEvent)
```

Comment:
Please restore `kube.RetryUntilSucceed` usage. Controller should keep retrying to watch clusters if error happens.

Reply:
Done.